proto/poktroll/supplier/tx.proto

```diff
@@ -45,17 +45,30 @@ message MsgUpdateParams {
 message MsgUpdateParamsResponse {}
 
 message MsgStakeSupplier {
-  option (cosmos.msg.v1.signer) = "address"; // https://docs.cosmos.network/main/build/building-modules/messages-and-queries
-  string address = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"]; // The Bech32 address of the supplier using cosmos' ScalarDescriptor to ensure deterministic deterministic encoding using cosmos' ScalarDescriptor to ensure deterministic deterministic encoding
-  cosmos.base.v1beta1.Coin stake = 2; // The total amount of uPOKT the supplier has staked. Must be ≥ to the current amount that the supplier has staked (if any)
-  repeated poktroll.shared.SupplierServiceConfig services = 3; // The list of services this supplier is staked to provide service for
+  option (cosmos.msg.v1.signer) = "sender"; // https://docs.cosmos.network/main/build/building-modules/messages-and-queries
+
+  string sender = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"]; // The Bech32 address of the message sender (i.e. owner or operator) using cosmos' ScalarDescriptor to ensure deterministic deterministic encoding using cosmos' ScalarDescriptor to ensure deterministic deterministic encoding
```

Comment:
I'm kind of confused by what a `sender` is.

There are two address:

1. Owner
2. Operator

Comments:
- No TODO about renaming address to operator_address
- No clue who the sener is

Reply:
This is a workaround the technical constraint which is about knowing who is signing the `StakeSupplierMsg`.

After `0.50` upgrade, there's no longer the `msg.GetSigners` method, so we don't know who signed the message without adding the `sender` field, making it the signer then determine whether it's the owner or the operator.

I'll add better comments describing what is a `sender` if we decide to keep it.

The other solution would be to have different messages each one signed by the corresponding entity.

Reply:
1. Instead of `sender`, it should be `signer` if we do keep it.
2. `The other solution would be to have different messages each one signed by the corresponding entity.` seems to complicated.
3. See details below....

----

I did a quick google search and found:

1. Bug discussing that `DefineCustomGetSigners is unusable` [1]
2. It references a PR [2] as a workaround.
3. The PR from (2) mentions two things:
- The correct way to do this is to use `cosmos.msg.v1.signer` which we already do in some parts of the codebase. More details in [3]
![Screenshot 2024-08-07 at 2 10 19 PM](https://github.com/user-attachments/assets/cdba19fb-154a-4df0-899d-af390f100316)
- Alternatively, we can create a custom signers function like so [4]

The options from (3) seem to be more idiomatic and correct, why not do that?


[1] https://github.com/cosmos/cosmos-sdk/issues/20077
[2] https://github.com/b-harvest/Canto/pull/62
[3] https://github.com/cosmos/cosmos-sdk/blob/release/v0.50.x/UPGRADING.md#all
[4] https://github.com/b-harvest/Canto/pull/62/files

Reply:
We'll opt for just renaming `sender` to `signer`